saito-core/src/core/io/network.rs

```diff
@@ -494,8 +494,15 @@ impl Network {
         let url;
         {
             let peers = self.peer_lock.read().await;
-
+            let wallet = self.wallet_lock.read().await;
             if let Some(peer) = peers.index_to_peers.get(&peer_index) {
+                if wallet.version > peer.version {
```

Comment:
better to do this in routing thread's process_incoming_block_hash also. move the function to Peer and call in both places.